advcpmv-0.8-8.32.patch

```diff
@@ -13,7 +14,57 @@ diff -aur coreutils-8.32/src/copy.c coreutils-8.32-patched/src/copy.c
 +  struct stat src_open_sb;
 +};
 +
-+/* Begin progress Mod*/
++char * escape_double_quotes ( char * _cUnescapedString )
++{
++  char * _cEscapedString;
++  _cEscapedString = (char *) calloc( sizeof(char) * strlen(_cUnescapedString) * 2, sizeof(char) );
++  size_t ipos = 0;
++  size_t rpos = 0;
++  for(rpos = 0, ipos = 0; _cUnescapedString[rpos] != '\0'; rpos++, ipos++)
++  {
++    if(_cUnescapedString[rpos] == '"')
++    {
++      _cEscapedString[ipos] = '\\';
++      ipos++;
++    }
++    if(_cUnescapedString[rpos] == '`')
++    {
++      _cEscapedString[ipos] = '\\';
++      ipos++;
++    }
++    _cEscapedString[ipos] = _cUnescapedString[rpos];
++  }
++  ipos++;
```

Comment:
Whatever valgrind error you were getting is because of this. The loop already incremented the ipos at termination. Remove this line.

Reply:
You can easily test it like this if you still have doubts:

```c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

char * escape_double_quotes ( char * _cUnescapedString )
{
	char * _cEscapedString;
	_cEscapedString = (char *) malloc( sizeof(char) * strlen(_cUnescapedString) * 2 );
	memset(_cEscapedString, 2, strlen(_cUnescapedString) * 2); //set everything to 2
	size_t ipos = 0;
	size_t rpos = 0;
	for(rpos = 0, ipos = 0; _cUnescapedString[rpos] != '\0'; rpos++, ipos++)
	{
		if(_cUnescapedString[rpos] == '"')
		{
			_cEscapedString[ipos] = '\\';
			ipos++;
		}
		if(_cUnescapedString[rpos] == '`')
		{
			_cEscapedString[ipos] = '\\';
			ipos++;
		}
		_cEscapedString[ipos] = _cUnescapedString[rpos];
	}
	++ipos;
	_cEscapedString[ipos] = '\0';
	for (size_t i = 0; i <= ipos; ++i)
		printf("%d", _cEscapedString[i]); //print out everything in int, it should print "1110" if done correctly
	fflush(stdout);
	return _cEscapedString;
}

int main(void)
{
	char test[] = { 1,1,1,0 };
	escape_double_quotes(test);
}
```

Reply:
By switching to your spawn function the escape function is obsolete and removed.